core-java-modules/core-java-collections-list-3/src/test/java/com/baeldung/list/arraylist/ListVsArrayListUnitTest.java

```diff
@@ -0,0 +1,72 @@
+package com.baeldung.list.arraylist;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertNotNull;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class ListVsArrayListUnitTest {
+
+    private final ByteArrayOutputStream outputStreamCaptor = new ByteArrayOutputStream();
+    private ArrayList<String> passengerList = new ArrayList<>();
+
+    @BeforeEach
+    public void setUp() throws Exception {
+        System.setOut(new PrintStream(outputStreamCaptor));
```

Comment:
Unit testing things that do System.out is pretty ugly. This test also fails to restore system.out afterwards.

You could use SystemStubs to handle this in Junit 5 - https://github.com/webcompere/system-stubs#using-systemerr-systemout-and-systemerrandout - where you can then use the `SystemOut` as a stub a bit like this - https://github.com/webcompere/system-stubs/blob/main/system-stubs-jupiter/src/test/java/uk/org/webcompere/systemstubs/jupiter/SystemStubsExtensionTest.java#L22 

But overall, why make code for an example that need tests this tricky to construct?

Reply:
Removed System.out statements and used return statements to test better.